### ISSUE TITLE
[Suggestion] Impliment "aim to cursor" controls

### ISSUE THREAD

**Issue by Noname117** [tsunamayo/Starship-EVO#2483]
One issue with the current mouse controls for the game while controlling ships is that the ships are very difficult to control. This comes down to the way the current mouse control scheme is set up.

In this game the mouse controls feature a center deadzone and moving the cursor outside of the deadzone causes the ship to rotate in that direction until the cursor is moved back into the deadzone. This sort of control scheme is unintuitive, particularly the part about moving the cursor back into the deadzone to stop rotating, and often leads to a lot of over-correction while making aiming difficult.

A much better control scheme is used in games like Starmade, Space Engineers, Avorion, and War Thunder, and can also be set up in Elite-Dangerous pretty easily, and is pretty much the standard for mouse controls in these sorts of games. In these games you move the cursor away from the center, and the ship rotates to the cursor, stopping when it reaches it (I guess technically the cursor gets moved to the center of the screen as the ship rotates). This is far more intuitive, since you're pointing to exactly where you want the ship aimed.

So is there any chance this control scheme could be implemented as an option? It would make the game far more controllable for us mouse-users.

**Comment by tsunamayo** [comment]
Yes I think you might be right, I will try this control scheme and check if there any issue with it.
Can I ask you how do you select things in space in those game?
I dont have time to play...
Thanks

**Comment by Noname117** [comment]
So to select things in those games, they use a couple of different methods. War Thunder and, if I recall correctly, Starmade have a hotkey to select stuff, and whatever you aim at gets selected. War Thunder has an added hotkey which, when held, allows you to look around without changing the direction your plane is flying in (although roll, yaw, and pitch keys still work), allowing you to select stuff you're not facing towards. Elite has part of the GUI dedicated to selecting objects, and usually has menus to sort through. I'd think in combat you could just aim at something and select it though.

I think some games have you cycle through targets you're aiming in the rough direction of when you press the hotkey which allows you to select an object. That is also an idea. Not sure if that would be intuitive or not, so some experimenting may need to be done here.

**Comment by ProPeach** [comment]
Most games implement a similar "point and click" way of selecting things as SEVO does currently, although some have an optional hotkey in the keybinds for things like "target nearest hostile" or "cycle next target". These are usually good enough, especially when tied to the smoother no use control system that Noname talks about.

**Comment by Tezlz** [comment]
I've only played war thunder and space engineers out of those games and afaik you can't select things in Space Engineers. In War Thunder however you can target another plane by middle mouse clicking and the game automatically selects the closest plane to your cursor so you don't have to look directly at the plane and accidentally turn that way.
Although that system might not work well in this game as you may not want to accidentally select the wrong thing because it happened to be closer to your cursor. You can probably solve this by making it so that you can select things in the 'tab' mode where you can look around without controling the ship, therefore you don't have to worry about accidentally turning the ship when you don't want it to.
